src/editors/containers/TextEditor/components/ImageSettingsModal/hooks.js

```diff
@@ -201,32 +233,43 @@ export const onCheckboxChange = (handleValue) => (e) => handleValue(e.target.che
  * @param {object} dimension - image dimensions ({ width, height })
  * @param {bool} isDecorative - is the image decorative?
  * @param {func} saveToEditor - save method for submitting image settings.
+ * @param {object} altTextError - error handling for alt-text
+ *   {func} setError - set isError to true
+ *   {func} dismissError - set isError to false
+ *   {func} setValid - set isValid to true
+ *   {func} setNotValid - set isValid to false
  */
 export const onSaveClick = ({
   altText,
   dimensions,
   isDecorative,
   saveToEditor,
-}) => () => saveToEditor({
-  altText,
-  dimensions,
-  isDecorative,
-});
+  altTextError,
+}) => () => {
+  let error = false;
 
-/**
- * isSaveDisabled(altText)
- * Returns true the save button should be disabled (altText is missing and not decorative)
- * @param {object} altText - altText hook object
- *   {bool} isDecorative - is the image decorative?
- *   {string} value - alt text value
- * @return {bool} - should the save button be disabled?
- */
-export const isSaveDisabled = (altText) => !altText.isDecorative && (altText.value === '');
+  // Alt Text Errors
+  if (!isDecorative && (altText === '')) {
+    altTextError.setError();
+    altTextError.setNotValid();
+    error = true;
+  } else {
+    altTextError.dismissError();
+    altTextError.setValid();
+  }
+
+  if (!error) {
+    saveToEditor({
+      altText,
+      dimensions,
+      isDecorative,
+    });
+  }
+};
```

Comment:
This is less concise than can be and is left over from when I tried to write an error for dimension controls. I think leaving it in this format could be useful if additional user feedback errors are added in the future.

Reply:
If that is the goal, then you should separate out that check into a `checkFormValidation` hook (which I think would be a good idea)

Reply:
^ +1

Reply:
I'll code that in. Ben and I also had a conversation on when the validation should be occurring (it should be occurring onChange for alt-text field and checkbox). I'll be coding that in as well.